identity/identity.go

```diff
@@ -519,10 +493,9 @@ func (i *Identity) WithDeclassifiedCredentials(ctx context.Context, c cipher.Pro
 					key := fmt.Sprintf("%d.%s", i, token)
 					ciphertext := v.Get(token).String()
 
-					var plaintext []byte
-					plaintext, err := c.Cipher(ctx).Decrypt(ctx, ciphertext)
-					if err != nil {
-						plaintext = []byte("")
+					plaintext, decryptErr := c.Cipher(ctx).Decrypt(ctx, ciphertext)
+					if decryptErr != nil {
+						plaintext = []byte{}
 					}
 					toPublish.Config, err = sjson.SetBytes(toPublish.Config, "providers."+key, string(plaintext))
 					if err != nil {
```

Comment:
The error here would not be correctly returned because it is a different scope than the one that is finally returned.